core/types/transaction_meta.go

```diff
@@ -7,19 +7,35 @@ package types
 import (
 	"bytes"
 	"encoding/binary"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+type QueueOrigin int64
+
+const (
+	// Possible `queue_origin` values
+	QueueOriginL1ToL2    QueueOrigin = 0
+	QueueOriginSafety    QueueOrigin = 1
+	QueueOriginSequencer QueueOrigin = 2
+)
+
 type TransactionMeta struct {
 	L1RollupTxId      *hexutil.Uint64   `json:"l1RollupTxId"`
 	L1MessageSender   *common.Address   `json:"l1MessageSender"`
 	SignatureHashType SignatureHashType `json:"signatureHashType"`
+	QueueOrigin       *big.Int          `json:"queueOrigin"`
 }
 
-func NewTransactionMeta(L1RollupTxId *hexutil.Uint64, L1MessageSender *common.Address, sighashType SignatureHashType) *TransactionMeta {
-	return &TransactionMeta{L1RollupTxId: L1RollupTxId, L1MessageSender: L1MessageSender, SignatureHashType: sighashType}
+func NewTransactionMeta(l1RollupTxId *hexutil.Uint64, l1MessageSender *common.Address, queueOrigin QueueOrigin, sighashType SignatureHashType) *TransactionMeta {
+	return &TransactionMeta{
+		L1RollupTxId:      l1RollupTxId,
+		L1MessageSender:   l1MessageSender,
+		QueueOrigin:       big.NewInt(int64(queueOrigin)),
```

Comment:
I'm not sure how I feel about this api, the `NewTransactionMeta` accepts a `QueueOrigin` which is a simple type wrapper around `int64` but it is internally represented as a `*big.Int`. I don't think it needs to be a `*big.Int`, but it serializes correctly as a `*big.Int` when passed to `abi.Pack`